warn/scrapers/ms.py

```diff
@@ -0,0 +1,116 @@
+from pathlib import Path
+
+import pdfplumber
+from bs4 import BeautifulSoup
+
+from .. import utils
+from ..cache import Cache
+
+__authors__ = ["Ash1R"]
+__tags__ = ["html", "pdf"]
+__source__ = {
+    "name": "Mississippi Department of Employment Security",
+    "url": "https://mdes.ms.gov/information-center/warn-information/",
+}
+
+
+def scrape(
+    data_dir: Path = utils.WARN_DATA_DIR,
+    cache_dir: Path = utils.WARN_CACHE_DIR,
+) -> Path:
+    """
+    Scrape data from Iowa.
+
+    Keyword arguments:
+    data_dir -- the Path were the result will be saved (default WARN_DATA_DIR)
+    cache_dir -- the Path where results can be cached (default WARN_CACHE_DIR)
+
+    Returns: the Path where the file is written
+    """
+    # Grab the page
+    page = utils.get_url("https://mdes.ms.gov/information-center/warn-information/")
+    html = page.text
+
+    # Write the raw file to the cache
+    cache = Cache(cache_dir)
+    cache.write("xx/yy.html", html)
+    soup = BeautifulSoup(html, "html5lib")
+    a = soup.select("a[href*=pdf]")
+    alllinks = ["https://mdes.ms.gov/" + link["href"] for link in a]
+    links = []
+    pdf_list = []
+    for link in alllinks:
+        if "map" not in link:
+            links.append(link)
+    for i in links:
+        cache_key = i.split("/")[-2] + i.split("/")[-1]
```

Comment:
What's going on here?

Reply:
I'm not exactly sure how to properly name the cache_keys, I tried to generate a unique key by taking the last two parts of every pdf link's url, which contains the month, year, and quarter.